### PR TITLE
Implement error handling middleware and improve some of the types on …

### PR DESCRIPTION
**Pull request by mattdean-digicatapult**
Implements error handling middleware. Note whilst I haven't included a specific integration test for this we don't really have any set up at the moment. I think we should track that separately and implement asap

### CODE REVIEW

File: src/controllers/scenario/index.ts [hunk 1]
```diff
@@ -43,15 +45,19 @@ ${eof}`
...
+        } : {
+          status: 'success',
           output: stdout,
-          ...err ? { exception: err } : {}
         })
       })
     })
   }
 
+  @Response<ValidateErrorJSON>(422, "Validation Failed")
```
Comment (n3op2) [inline src/controllers/scenario/index.ts:60]: Yes! This is wahat I wanted, but have not checked any docs or anything regarding this

Comment (n3op2) [reply src/controllers/scenario/index.ts:60]: Wonder how much we can expand on this e.g. getting error instance to return some nice formatted stuff back to the client.

**Review by n3op2 — APPROVED** [review]

File: src/controllers/scenario/index.ts (hunk 1 above)
Comment (n3op2) [reply src/controllers/scenario/index.ts:60]: What response looks like? of this 422

Comment (mattdean-digicatapult) [reply src/controllers/scenario/index.ts:60]: It's actually more yucky than I'd like for example wit ha bad executable:

```json
{
  "message": "Validation Failed",
  "details": {
    "executable": {
      "message": "Could not match the union against any of the items. Issues: [{\"executable\":{\"message\":\"should be one of the following; ['out-of-bounds-access-aarch64']\",\"value\":\"ou-of-bounds-read-aarch64\"}},{\"executable\":{\"message\":\"should be one of the following; ['out-of-bounds-access-cheri']\",\"value\":\"ou-of-bounds-read-aarch64\"}},{\"executable\":{\"message\":\"should be one of the following; ['out-of-bounds-read-aarch64']\",\"value\":\"ou-of-bounds-read-aarch64\"}},{\"executable\":{\"message\":\"should be one of the following; ['out-of-bounds-read-cheri']\",\"value\":\"ou-of-bounds-read-aarch64\"}},{\"executable\":{\"message\":\"should be one of the following; ['out-of-bounds-readV2-aarch64']\",\"value\":\"ou-of-bounds-read-aarch64\"}},{\"executable\":{\"message\":\"should be one of the following; ['out-of-bounds-readV2-cheri']\",\"value\":\"ou-of-bounds-read-aarch64\"}},{\"executable\":{\"message\":\"should be one of the following; ['out-of-bounds-write-aarch64']\",\"value\":\"ou-of-bounds-read-aarch64\"}},{\"executable\":{\"message\":\"should be one of the following; ['out-of-bounds-write-cheri']\",\"value\":\"ou-of-bounds-read-aarch64\"}},{\"executable\":{\"message\":\"should be one of the following; ['use-after-free-aarch64']\",\"value\":\"ou-of-bounds-read-aarch64\"}},{\"executable\":{\"message\":\"should be one of the following; ['use-after-free-cheri']\",\"value\":\"ou-of-bounds-read-aarch64\"}}]",
      "value": "ou-of-bounds-read-aarch64"
    }
  }
}
```

But unfortunately that's what tsoa produces for the validation error. My implementation btw is modified from https://tsoa-community.github.io/docs/error-handling.html#setting-up-error-handling

**Pull request merged**